hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -621,8 +635,14 @@ private void initializeFileGroups(HoodieTableMetaClient dataMetaClient, Metadata
 
     LOG.info(String.format("Creating %d file groups for partition %s with base fileId %s at instant time %s",
         fileGroupCount, metadataPartition.getPartitionPath(), metadataPartition.getFileIdPrefix(), instantTime));
+    HoodieTableFileSystemView fsView = HoodieTableMetadataUtil.getFileSystemView(metadataMetaClient);
+    List<FileSlice> fileSlices = HoodieTableMetadataUtil.getPartitionLatestFileSlices(metadataMetaClient, Option.ofNullable(fsView), metadataPartition.getPartitionPath());
     for (int i = 0; i < fileGroupCount; ++i) {
       final String fileGroupFileId = String.format("%s%04d", metadataPartition.getFileIdPrefix(), i);
+      // if a writer or async indexer had already initialized the filegroup then continue
+      if (!fileSlices.isEmpty() && fileSlices.stream().anyMatch(fileSlice -> fileGroupFileId.equals(fileSlice.getFileGroupId().getFileId()))) {
```

Comment:
We should not get into this case. Can you please explain how this can happen? Its either all filegroups are initialized or nothing is initialized. This block gives a sense that it can be partially initialized and the incomplete filgroups are inited here. Which shouldn't happen right?

Reply:
With  initialization happening while scheduling index, we should not get into this case anymore.
> Ideally initializeFileGroups should be called just once per MDT partition right ?or am I missing something.

That's correct.